.github/workflows/ci.yml

```diff
@@ -76,33 +76,62 @@ jobs:
     outputs:
       os: ${{ steps.setup-os-matrix.outputs.os }}
 
-  build-and-test:
-    needs: setup-os-matrix
+  build:
+    needs: check-style
```

Comment:
`build` can run in parallel to `check-style` (see reasoning in other comments below)
```suggestion
```